client/state-db/src/pruning.rs

```diff
@@ -202,108 +488,109 @@ impl<BlockHash: Hash, Key: Hash> RefWindow<BlockHash, Key> {
 		self.pending_canonicalizations = 0;
 		for _ in 0..self.pending_prunings {
 			let pruned = self
-				.death_rows
-				.pop_front()
-				.expect("pending_prunings is always < death_rows.len()");
+				.queue
+				.pop_front(self.pending_number)
+				// NOTE: `pop_front` should not return `MetaDb::Error` because blocks are visited
+				// by `RefWindow::prune_one` first then `RefWindow::apply_pending` and
+				// `DeathRowQueue::get` should load the blocks into cache already
+				.expect("block must loaded in cache thus no MetaDb::Error")
```

Comment:
The NOTE does not make much sense to me, for me we just do not expect to have a missing journal and queue (queue and uncached) to be empty if pending_pruning is 0. (and pending pruning to be out of queue but this is next expectaiton)

Reply:
`pop_front` return `Result<Option<DeathRow<BlockHash, Key>>, Error<D::Error>>`, the first `expect` is for the potential loading from db thus brings potential `Error<D::Error>` which is unexpected because there should no db loading for `pop_front` in the current usage as the comment explained. And the second `expect` is for the case you mentioned.